src/elexmodel/models/BaseElectionModel.py

```diff
@@ -1,94 +1,40 @@
-        # in that case, we catch the error and warning and re-run with normalize_weights false
-        try:
-            model.fit(
-                X,
-                y,
-                tau_value=tau,
-                weights=weights,
-                lambda_=self.lambda_,
-                fit_intercept=self.add_intercept,
-                normalize_weights=normalize_weights,
-            )
-        except (UserWarning, cvxpy.error.SolverError):
-            LOG.warning("Warning: solution was inaccurate or solver broke. Re-running with normalize_weights=False.")
-            model.fit(X, y, tau_value=tau, weights=weights, lambda_=self.lambda_, normalize_weights=False)
-
-    def get_unit_predictions(self, reporting_units, nonreporting_units, estimand):
+    def get_unit_predictions(
+        self, reporting_units: pd.DataFrame, nonreporting_units: pd.DataFrame, estimand: str, *kwargs
+    ) -> np.ndarray:
         """
-        Produces unit level predictions. Fits quantile regression to reporting data, applies
-        it to nonreporting data. The features are specified in model_settings.
+        Generates and returns unit level predictions
         """
-        # compute the means of both reporting_units and nonreporting_units for centering (part of featurizing)
-        # we want them both, since together they are the subunit population
-        self.featurizer.compute_means_for_centering(reporting_units, nonreporting_units)
-        # reporting_units_features and nonreporting_units_features should have the same
-        # features. Specifically also the same fixed effect columns.
-        reporting_units_features = self.featurizer.featurize_fitting_data(
-            reporting_units, add_intercept=self.add_intercept
-        )
-        nonreporting_units_features = self.featurizer.featurize_heldout_data(nonreporting_units)
-
-        weights = reporting_units[f"last_election_results_{estimand}"]
-        reporting_units_residuals = reporting_units[f"residuals_{estimand}"]
-
-        self.fit_model(self.qr, reporting_units_features, reporting_units_residuals, 0.5, weights, True)
-        self.features_to_coefficients = dict(zip(self.featurizer.complete_features, self.qr.coefficients))
-
-        preds = self.qr.predict(nonreporting_units_features)
-
-        # multiply by total voters to get unnormalized residuals
-        preds = preds * nonreporting_units[f"last_election_results_{estimand}"]
-
-        # add in last election results to go from residual to number of votes in this election
-        # max with results so that predictions are always at least ars large as actual number of votes
-        preds = np.maximum(
-            preds + nonreporting_units[f"last_election_results_{estimand}"], nonreporting_units[f"results_{estimand}"]
-        )
-
-        # round since we don't need the artificial precision
-        return preds.round(decimals=0)
+        pass
```

Comment:
Is the idea that the `BaseElectionModel` can't be instantiated and one has to use a subclass instead?  Because in that case, I'd recommend creating it as an abstract base class and have this (and similar) method(s) return `NotImplemented` so that an exception is raised: https://docs.python.org/3/library/abc.html 🤔

Reply:
yes, that is the idea! That's a great idea, will do -- thanks!